src/components/Insights/ActivityFeed.js

```diff
@@ -52,7 +67,18 @@ export default class ActivityFeed extends React.Component {
         const externalsourceid = props.externalsourceid !== constants.DEFAULT_EXTERNAL_SOURCE ? props.externalsourceid : null;
         const fulltextTerm = "";
 
-        SERVICES.FetchMessageSentences(externalsourceid, bbox, zoomLevel, fromDate, toDate, ActivityConsts.OFFSET_INCREMENT, pageState, [maintopic].concat(Array.from(termFilters)), pipelinekeys, fulltextTerm, callback);
+        ADMIN_SERVICES.fetchTrustedSources(pipelinekeys, "", (trustedSourcesErr, trustedSourcesResponse, trustedSourcesBody) => {
```

Comment:
Move the trusted source service lookup to the dashboard action and bind to the DataStore. We want to avoid this as this lookup will by invoked for each page scroll in the news feed component.

Reply:
I wanted to isolate this change as much as possible in order to reduce the risk, but since there is already another place that reaches into the admin services to make this same call, maybe it's indeed time to pull this up.

Reply:
Alright, @erikschlegel, I pushed the trusted sources fetching out of the ActivityFeed component into the dashboard initialization (so we only do it once) in 33ae6ad and then also refactored the TypeaheadSearch to avoid re-fetching the trusted sources in ed56620.